### ISSUE TITLE
indexing speed dropped by huge factor after upgrade

### ISSUE THREAD

**Issue by iahmad-khan** [ip2location/logstash-filter-ip2location#18]
Just upgraded to latest version , the indexing speed dropped by a factor of 30. It is now 30 times slow to process data. Can you please check and investigate which might be cuasing this.

**Comment by iahmad-khan** [comment]
Can someone have a look at this issue?

**Comment by iahmad-khan** [comment]
As shown , the latest version was struggling to index the data for last 30 hours and still doing it , by switching to previous buggy version it indexed the data in just a few seconds. 

The previous version is buggy but very fast , maybe the new version is single threaded or something.
![Screenshot from 2019-12-04 13-05-08](https://user-images.githubusercontent.com/13478264/70145028-14058400-1697-11ea-8573-43c0ed704322.png)

the indexing rate of the latest version is so low that it is shown 0 as compared to the old one

**Comment by ip2location-com** [comment]
Hi @iahmad-khan 

Thank you for your feedback. Can you show us your Configuration file?

**Comment by iahmad-khan** [comment]
Which configuration file? Logstash configuration file is unchanged. Do I
need to change it

**Comment by iahmad-khan** [comment]
```
ip2location {
    source => "clientip"
    use_memory_mapped => true
    database => "/usr/share/logstash/ip2location.BIN"
  }


```

**Comment by ip2location-com** [comment]
Hi @iahmad-khan 

We are not able to reproduce the issue. 
Based on our testing, about 80 seconds are used to process 50k of data and we are using the commercial IP2Location DB11 Database.

**Comment by iahmad-khan** [comment]
When I revert back to the previous version , the issue goes away , so I am sure this is related to the new version.

**Comment by iahmad-khan** [comment]
Hi, I am still facing this issue becuase I need to upgrade to logstash 7.5 and need to install the plguin that brings the latest one , which has the problem of indexing speed.  Infact it stucks forever and sends nothing to ES. When I remove the ip2location filter , it works and sends all data in just few minutes.

Machine: 32 GB ram , 8 CPU , with heap size assigned: 24 Gb 

Here is my log-stash config file 


```

input {
  beats {
    port => "5041"
    client_inactivity_timeout => 40000
  }
}

filter {
  grok {
    match => { "message" => "%{DATA:endpoint}	%{IP:clientip}	%{NUMBER:count_requests}	%{NUMBER:sum_size}	%{NUMBER:http_200_count}	%{NUMBER:http_300_count}	%{NUMBER:http_400_count}	%{NUMBER:http_500_count}	%{NUMBER:http_unkown_count}	%{NUMBER:pages_count}	%{NUMBER:robot_count}"}
    remove_field => ["message"]
  }


  ip2location {
    source => "clientip"
    use_memory_mapped => true
    database => "/usr/share/logstash/ip2location.BIN"
  }

  mutate {
     add_field => { "[location][lat]" => "%{[ip2location][latitude]}" }
     add_field => { "[location][lon]" => "%{[ip2location][longitude]}" }
  }

  mutate {
     convert => {
       "[location][lat]" => "float"
       "[location][lon]" => "float"
    }
  }

}

output {
  elasticsearch {
    hosts => [ "localhost:9200" ]
    index => "weblogs-%{+YYYY}"
  }
}


```

Try this on a log file with for example 5 million lines and let me know.

**Comment by amey55** [comment]
+1 
1. I am facing the same problem, with ip2location plugin version 2.0.2 the performance was good but as soon as I upgraded the plugin to version 2.1.1. the indexing speed dropped and it took almost 18 hrs to process only 300,000 events, In config part, this field is also present:  use_memory_mapped => true
2. Also, the reason I had to upgrade the plugin was [this](https://github.com/ip2location/logstash-filter-ip2location/issues/15) error.
3. In version 2.0.2 I was getting the "wrong number of arguments (1 for 3)" error" and coz of that the logstash was crashing once a day at some fixed time.
